hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/grid/AbstractGrid.java

```
@@ -52,56 +82,595 @@ protected AbstractGrid(final VaadinMessageSource i18n, final UIEventBus eventBus
+        public String getStyle(final CellReference cellReference) {
+            if (center != null && Arrays.asList(center).contains(cellReference.getPropertyId())) {
+                return "centeralign";
+            } else if (right != null && Arrays.asList(right).contains(cellReference.getPropertyId())) {
+                return "rightalign";
+            } else if (left != null && Arrays.asList(left).contains(cellReference.getPropertyId())) {
+                return "leftalign";
+            }
+            return null;
+        }
+    }
+
+    /**
+     * Adds a tooltip to the 'Date and time' column in detailed format.
+     */
+    public static class ModifiedTimeTooltipGenerator implements CellDescriptionGenerator {
+        private static final long serialVersionUID = -6617911967167729195L;
+
+        private final String datePropertyId;
+
+        /**
+         * Constructor.
+         *
+         * @param datePropertyId
+         */
+        public ModifiedTimeTooltipGenerator(final String datePropertyId) {
+            this.datePropertyId = datePropertyId;
+        }
+
+        @Override
+        public String getDescription(final CellReference cell) {
+            String tooltip = null;
+            if (datePropertyId.equals(cell.getPropertyId())) {
+                final Long timestamp = (Long) cell.getItem().getItemProperty(datePropertyId).getValue();
+                tooltip = SPDateTimeUtil.getFormattedDate(timestamp);
+            }
+            return tooltip;
+        }
+    }
+
+    /**
+     * Converter that gets time-data as input of type <code>Long</code> and
+     * converts to a formatted date string.
+     */
+    public class LongToFormattedDateStringConverter implements Converter<String, Long> {
+        private static final long serialVersionUID = 1247513913478717845L;
+
+        @Override
+        public Long convertToModel(final String value, final Class<? extends Long> targetType, final Locale locale) {
+            // not needed
```

Then it's better to throw a `UnsupportedOperationException`